src/hooks/users.ts

```diff
@@ -1,71 +1,106 @@
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
 
-// NOTE: Heuristic approach. Way faster and efficient.
-// Is based on assumption that there is only one parent venue and it is an entry point for any user
+// NOTE: Logically correct version of pulling users
 export const useConnectUniverseUsers = () => {
   const venueId = useVenueId();
+  const firestore = useFirestore();
 
-  const { currentVenue, isCurrentVenueLoaded } = useConnectCurrentVenueNG(
-    venueId
-  );
-
-  useSparkleFirestoreConnect(() => {
-    if (!isCurrentVenueLoaded || !currentVenue || !venueId) return [];
-
-    const queryVenueId = currentVenue.parentId || venueId;
-    return [
-      {
-        collection: "users",
-        where: ["enteredVenueIds", "array-contains", queryVenueId],
-        storeAs: "universeUsers",
-      },
-    ];
-  });
-};
-
-// NOTE: Logically correct version of pulling users
-export const useConnectUniverseUsers_V2 = () => {
-  const venueId = useVenueId();
+  const { currentVenue, isCurrentVenueLoaded } = useConnectCurrentVenueNG();
 
   const { relatedVenues, isRelatedVenuesLoaded } = useConnectRelatedVenues({
     venueId,
   });
 
-  useSparkleFirestoreConnect(() => {
+  // useEffect(() => {
+  //   if (!isRelatedVenuesLoaded) return;
+
+  //   const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
+
+  //   const chunkedListeners = chunk(universeUserVenueIds, 10).map<
+  //     SparkleRFQConfig
+  //   >((venueIdsChunk) => ({
+  //     collection: "users",
+  //     where: ["enteredVenueIds", "array-contains", venueIdsChunk[0]],
+  //     storeAs: "universeUsers",
+  //   }));
+
+  //   firestore.setListeners(chunkedListeners);
+  // }, [isRelatedVenuesLoaded, relatedVenues]);
+
+  // const usersQuery = useMemo<SparkleRFQConfig | undefined>(() => {
+  //   if (!isRelatedVenuesLoaded || !venueId) return;
+
+  //   const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
+
+  //   console.log("venues", relatedVenues);
+
+  //   return {
+  //     collection: "users",
+  //     where: ["enteredVenueIds", "array-contains-any", universeUserVenueIds],
+  //     storeAs: "universeUsers",
+  //   };
+  // }, [venueId, isRelatedVenuesLoaded]);
+
+  // useSparkleFirestoreConnect(usersQuery);
+
+  // if (!isRelatedVenuesLoaded || !venueId) return;
+
+  const queryFn = (): SparkleRFQConfig[] => {
     if (!isRelatedVenuesLoaded) return [];
 
-    const universeUserVenueIds = relatedVenues.map((venue) =>
-      venue.name.toLowerCase()
-    );
+    const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
 
-    return universeUserVenueIds.map((venueId) => ({
-      collection: "users",
-      where: ["enteredVenueIds", "array-contains", venueId],
-      storeAs: "universeUsers",
-    }));
-  });
+    return [
+      {
+        collection: "users",
+        where: ["enteredVenueIds", "array-contains", universeUserVenueIds[1]],
+        storeAs: "universeUsers",
+      },
+    ];
+  };
```

Comment:
Same issues as called out in https://github.com/sparkletown/sparkle/pull/1047/commits/2cf34bcbe560aba82f7c3dde33d164cbd437e6bd#r554650801 and TODOADDLINK